bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/ToolBar.java

```diff
@@ -735,6 +735,46 @@ void layoutItems () {
 			}
 		}
 	}
+	
+	/*
+	 * Feature in Windows. When a tool bar has style SWT.FLAT, SWT.HORIZONTAL and
+	 * contains both text and images, Window leaves too much padding around the button.
+	 * This affects every button in the tool bar and makes the preferred height too big.
+	 * The fix is to set the TBSTYLE_LIST when the tool bar contains both text and images.
+	 */
+	if (OS.IsAppThemed()) {
+		if ((style & SWT.FLAT) != 0 && (style & SWT.HORIZONTAL) != 0) {
+			boolean hasText = false, hasImage = false;
+			for (ToolItem item : items) {
```

Comment:
if a SINGLE toolbar item has both image + text we do not want to do this condensing.